day-4/day-4.dylan

```diff
@@ -44,37 +44,37 @@ define function check-win?(board :: <sequence>) => (win :: <boolean>)
 end;
 
 define function score-board(board :: <sequence>) => (score :: <integer>)
-  let value-or-0 = method (i) if (i = #t) 0 else i end; end;
-  let reduce-row = method (row) reduce1(\+, map(\value-or-0, row)) end;
-  reduce1(\+, map(\reduce-row, board));
+  local
```

Comment:
I'm guessing local means these are only bound in the below extant?

Reply:
Yeah. It has three benefits over plain `let`:

1. The methods within a single `local` form can call each other. With `let` the first one can't call the second one. (I realize you don't need this, but see the 3rd point.)
2. Better formatting (less indentation for the method bodies).
3. It's the predominant style for writing local functions.

If you like making a distinction between `flet` and `labels` you can do this for `flet` semantics:

```dylan
local method one () ... end;
local method two () ... end;
```